src/include/common/object_pool.h

```diff
@@ -55,12 +70,25 @@ class AlignedByteAllocator {
 template <typename T, class Allocator = AlignedByteAllocator<T>>
 class ObjectPool {
  public:
-  /**
+  /***
    * Initializes a new object pool with the supplied limit to the number of
    * objects reused.
-   * @param reuse_limit
+   *
+   * @param size_limit the maximum number of objects the object pool controls
+   * @param reuse_limit the maximum number of reusable objects, which needs be to
+   * not greater than size_limit
    */
-  explicit ObjectPool(const uint64_t reuse_limit) : reuse_limit_(reuse_limit) {}
+  explicit ObjectPool(const uint64_t size_limit, const uint64_t reuse_limit)
+      : size_limit_(size_limit), reuse_limit_(reuse_limit), current_size_(0) {}
+
+  /**
+   * Initializes a new object pool with the supplied limit to the number of
+   * objects reused. The reuse_limit is set to be the same as size limit.
+   *
+   * @param size_limit the number of objects the object pool controls
+   */
+  explicit ObjectPool(const uint64_t size_limit)
```

Comment:
This is basically a default argument. I say we remove it.

Reply:
@pervazea  check-clang-tidy requires that every single argument constructor needs to be marked as explicit.

Reply:
I was more suggesting that we always take in an explicit reuse_limit as second argument